### PR TITLE
PR: Fix tab does not switch when ctrl is released issue

### LINKED ISSUE #4763: In the Editor, tab does not switch when ctrl is released and focus remains on the tab_switcher
## Description of your problem

**What steps will reproduce the problem?**

1. While focus is in the Editor, hold ctrl key and click Tab.
2. Release ctrl key

**What is the expected output? What do you see instead?**

_Expected output:_ When ctrl key is released, Editor tab should change to the one selected and tab_switcher should close.

_Actual output:_ When ctrl key is released, Editor tab does not change and focus remains on the tab_switcher. A mouse double-click on the tab_switcher is required for it to switch tab and to close.

**Please provide any additional information below**

## Versions and main components

* Spyder Version: 3.2.0.dev0
* Python Version: 3.6.1
* Qt Version: 4.8.7
* PyQt Version: 4.11.4
* Operating system: Win7 64bits

## Dependencies

IPython >=4.0    :  6.1.0 (OK)
cython >=0.21    :  None (NOK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
numpy >=1.7      :  1.13.0 (OK)
pandas >=0.13.1  :  None (NOK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.7.1 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.6.2 (OK)
sympy >=0.7.3    :  None (NOK)

tab_switcher

### PR DESCRIPTION
**Pull request by jnsebgosselin**
Fixes #4763

----

It seems like the ctrl key release event never reach the [EditorStack ](https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/editor.py). It seems to be swallowed by the [CodeEditor](https://github.com/spyder-ide/spyder/blob/master/spyder/widgets/sourcecode/codeeditor.py).

The proposed solution to solve this issue it to install an eventFilter on each `editor` added to
the `EditorStack` and to move the code that is in the `keyReleaseEvent` in the `eventFilter` method.

There may be a more elegant way to solve this directly in the CodeEditor class. I can look into it if this approach is preferred.

### CODE REVIEW

**Comment by jnsebgosselin** [comment]
Ok, I've found another way to solve this issue that is better I think. We can add event.ignore() in keyReleaseEvent method of CodeEditor class so that the event is propagated to the parent. I think it is a better approach than using an eventFilter in the parent. 

The PR was modified accordingly.

**Comment by ccordoba12** [comment]
@rlaverde, please review this one.

File: spyder/widgets/editor.py [hunk 1]
```diff
@@ -2185,7 +2185,7 @@ def keyReleaseEvent(self, event):
                         self.tabs_switcher = None
                         return
 
-        super(EditorStack, self).keyPressEvent(event)
+        super(EditorStack, self).keyReleaseEvent(event)
```
Comment (rlaverde) [inline spyder/widgets/editor.py]: My fault, I introduced this typo, sorry 😞

File: spyder/widgets/sourcecode/codeeditor.py [hunk 2]
```diff
@@ -2746,6 +2746,7 @@ def keyReleaseEvent(self, event):
         """Override Qt method."""
         self.timer_syntax_highlight.start()
         super(CodeEditor, self).keyReleaseEvent(event)
+        event.ignore()
```
Comment (rlaverde) [inline spyder/widgets/sourcecode/codeeditor.py]: This was the error, In qt4 the event wasn't propagated to the parent (I had tested it with pyqt5)

**Pull request merged**